go/ai-proxy/proxy/ollama/ollama.go

```diff
@@ -0,0 +1,314 @@
+package ollama
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	ollamaapi "github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/openai"
+	"github.com/pluralsh/console/go/ai-proxy/api"
+	"k8s.io/klog/v2"
+)
+
+const (
+	ChatCompletionObject      = "chat.completion"
+	ChatCompletionChunkObject = "chat.completion.chunk"
+)
+
+type OllamaProxy struct {
+	toolRegistry *ToolRegistry
+	ollamaClient *ollamaapi.Client
+	api.OpenAIProxy
+}
+
+// ToolRegistry manages a collection of tools for efficient lookup
+type ToolRegistry struct {
+	tools       map[string]ollamaapi.ToolCall
```

Comment:
instead of having three hash maps, you should probably make an entry struct with three fields like `ToolEntry` and make the registry store a map[string]ToolEntry (that way you don't have to defensively program around the key being in one map vs the others).